core/src/main/java/io/aiven/kafka/tieredstorage/chunkmanager/DiskBasedCachingChunkManager.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.chunkmanager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import com.github.benmanes.caffeine.cache.Weigher;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.nio.file.StandardCopyOption.ATOMIC_MOVE;
+
+public class DiskBasedCachingChunkManager extends CachingChunkManager<Path> {
+
+    private static final Logger log = LoggerFactory.getLogger(DiskBasedCachingChunkManager.class);
+
+    private DiskBasedCachingChunkManagerConfig config;
+
+    public DiskBasedCachingChunkManager(final ChunkManager chunkManager, final Executor executor) {
+        super(chunkManager, executor);
+    }
+
+    @Override
+    public InputStream chunkToStream(final Path cachedChunk) {
+        try {
+            return Files.newInputStream(cachedChunk);
+        } catch (final IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Path cacheChunk(final ChunkKey chunkKey, final InputStream chunk) throws IOException {
+        final Path tempChunkPath = config.cacheTempPath().resolve(chunkKey.getUuid().toString())
+                .resolve(String.valueOf(chunkKey.getChunkId()));
+        Files.createDirectories(tempChunkPath.getParent());
+        final Path tempCached = Files.write(tempChunkPath, IOUtils.toByteArray(chunk));
```

Comment:
Wondering if readAllBytes is enough?
```suggestion
        final Path tempCached = Files.write(tempChunkPath, chunk.readAllBytes());
```

Reply:
Fixed.